<!-- Entry point for using OpenNI2 devices -->
<launch>
    <node pkg="rta_openni" type="rta_openni_node" name="camera" output="screen"/>
    <!-- launch-prefix="gdb -ex run"/>-->

    <!-- "camera" should uniquely identify the device. All topics are pushed down
         into the "camera" namespace, and it is prepended to tf frame ids. -->
    <arg name="camera_name" default="camera"/>
    <arg name="depth_ir_rgb_calibration_file_path" default="$(find rta_openni)/assets/calibration_depth_ir_rgb.xml"/>
    <arg name="ir_arm_calibration_file_path" default="$(find rta_openni)/assets/calibration_ir_arm.xml"/>
    <arg name="depth_displacement_file_path" default="$(find rta_openni)/assets/depth_displacement.xml"/>
    <arg name="tf_prefix" default=""/>
    <arg name="housing" default="/left_hand_camera"/>

    <arg name="pi/2" value="1.5707963267948966"/>
    <arg name="optical_rotate2" value="0 0 0 -$(arg pi/2) 0 -$(arg pi/2)"/>
    <arg name="optical_rotate" value="0 0 0 0 0 0"/>

    <!-- relationships of camera_link to robot frame, wherever it may be attached-->
    <!-- no need to publish very frequently, these transforms are static-->
    <!--<node pkg="tf" type="static_transform_publisher" name="$(arg camera)_base_link"
          args="-0.03951829 -0.12307098 -0.05177354 0.03437356 -0.00427629 -0.00637571 $(arg housing) $(arg tf_prefix)/$(arg camera)_link 10" />-->

    <!-- relationships of cameras to camera_link -->
    <node pkg="tf" type="static_transform_publisher" name="$(arg camera_name)_base_to_depth"
          args="0 0 0 0 0 0 $(arg tf_prefix)/$(arg camera_name)_link $(arg tf_prefix)/$(arg camera_name)_depth_frame 10"/>
    <node pkg="tf" type="static_transform_publisher" name="$(arg camera_name)_depth_to_optical_link"
          args="$(arg optical_rotate) $(arg tf_prefix)/$(arg camera_name)_depth_frame $(arg tf_prefix)/$(arg camera_name)_depth_optical_frame  10"/>
    <node pkg="tf" type="static_transform_publisher" name="$(arg camera_name)_color_to_optical_link"
          args="$(arg optical_rotate) $(arg tf_prefix)/$(arg camera_name)_color_frame $(arg tf_prefix)/$(arg camera_name)_color_optical_frame 10"/>

    <!-- parameters -->
    <param name="camera_name" type="str" value="$(arg camera_name)"/>
    <param name="tf_prefix" type="str" value="$(arg tf_prefix)"/>
    <param name="housing" type="str" value="$(arg housing)"/>
    <param name="depth_ir_rgb_calibration_file_path" type="str" value="$(arg depth_ir_rgb_calibration_file_path)"/>
    <param name="ir_arm_calibration_file_path" type="str" value="$(arg ir_arm_calibration_file_path)"/>
    <param name="depth_displacement_file_path" type="str" value="$(arg depth_displacement_file_path)"/>

</launch>
